src/game-data/visitors/summerVisitorReducers.tsx

```diff
@@ -776,8 +837,130 @@ export const summerVisitorReducers: Record<
         }
     },
     // planner: s => endVisitor(s),
-    // planter: s => endVisitor(s),
+    planter: (state, action, pendingAction) => {
+        const planterAction = pendingAction as PlayVisitorPendingAction & {
+            isDiscarding?: boolean;
+            secondPlant?: boolean;
+        };
+        switch (action.type) {
+            case "CHOOSE_CARDS":
+                if (!action.cards) {
+                    // pass on second plant
+                    return endVisitor(gainCoins(1, state));
+                }
+                const card = action.cards![0];
+                switch (card.type) {
+                    case "visitor":
+                        return promptForAction(state, {
+                            choices: [
+                                {
+                                    id: "PLANTER_PLANT",
+                                    label: <>Plant up to 2 <Vine /> and gain <Coins>1</Coins></>,
+                                    disabledReason: plantVinesDisabledReason(state)
+                                },
+                                {
+                                    id: "PLANTER_UPROOT",
+                                    label: <>Uproot and discard 1 <Vine /> to gain <VP>2</VP></>,
+                                    disabledReason: uprootDisabledReason(state)
+                                }
+                            ]
+                        });
+                    case "vine":
+                        return planterAction.isDiscarding ?
+                            endVisitor(gainVP(2, discardCards(action.cards!, state))) :
+                            promptToPlant(state, card.id);
+                    default:
+                        return state;
+                }
+            case "CHOOSE_ACTION":
+                switch (action.choice) {
+                    case "PLANTER_PLANT":
+                        return promptToChooseVineCard(state);
+                    case "PLANTER_UPROOT":
+                        return promptToUproot(state);
+                    default:
+                        return state;
+                }
+            case "CHOOSE_VINE":
+                state = uprootVineFromField(action.vines[0], state);
+                return promptToChooseCard(setPendingAction({
+                    ...planterAction,
+                    isDiscarding: true
+                }, state), {
+                    title: "Discard a vine",
+                    cards: state.players[state.currentTurn.playerId].cardsInHand
+                        .filter(({ type }) => type === "vine")
+                        .map(id => ({ id }))
+                });
+            case "CHOOSE_FIELD":
+                state = plantVineInField(action.fields[0], state);
+                const canPlantAgain = !planterAction.secondPlant &&
+                    plantVinesDisabledReason(state) === undefined;
+                return canPlantAgain ? promptToChooseVineCard(
```

Comment:
```ts
return canPlantAgain
    ? promptToChooseVineCard(
        ...
    )
    : endVisitor(gainCoins(1, state));
```

is there some setting for the autoformatter you're using? I find ternaries pretty hard to read if the `?`/`:` isn't aligned.